XenModel/WLB/WlbScheduledTask.cs

```diff
@@ -690,22 +690,26 @@ public WlbScheduledTasks(Dictionary<string,string> Configuration)
                 Dictionary<string, string> taskDictionary = scheduleTask.ToDictionary();
                 foreach (string key in taskDictionary.Keys)
                 {
-                    if (null == collectionDictionary)
+                    if (collectionDictionary == null)
                     {
                         collectionDictionary = new Dictionary<string, string>();
                     }
                     collectionDictionary.Add(key, taskDictionary[key]);
                 }
-                foreach (string key in scheduleTask.TaskParameters.Keys)
+
+                if (collectionDictionary != null)
                 {
-                    string complexKey = string.Format("{0}_{1}_{2}", scheduleTask.KeyBase, scheduleTask.TaskId.ToString(), key);
-                    if (collectionDictionary.ContainsKey(complexKey))
-                    {
-                        collectionDictionary[complexKey] = scheduleTask.TaskParameters[key];
-                    }
-                    else
+                    foreach (var key in scheduleTask.TaskParameters.Keys)
                     {
-                        collectionDictionary.Add(complexKey, scheduleTask.TaskParameters[key]);
+                        var complexKey = $"{scheduleTask.KeyBase}_{scheduleTask.TaskId}_{key}";
+                        if (collectionDictionary.ContainsKey(complexKey))

```

Comment:
is this needed?
Cant you just say collectionDictionary[complexKey] = scheduleTask. 
If the key doesnt exist then it will be created.